### PR TITLE
feat(create-medusa-app): add install Next.js storefront option

### PR DESCRIPTION
**Pull request by shahednasser**
## What

- Adds the ability to install the Next.js storefront along with the Medusa backend when using the `create-medusa-app` command. This can be done either through passing the `--with-nextjs-starter` option or answering "y" when asking about installing the Next.js starter.

**Note: The branch cloned of the medusa starter default should be removed before merging this PR**

### With Next.js Starter Without Option

1. Run the command (either locally or through the snapshot).
2. When asked about the Next.js starter, answer "y" (default is "N")
3. Wait for the installation steps to finish, including the Next.js starter. The starter will be installed under the `[project-name]-storefront` directory. `[project-name]` is the name you provide to the project in the first question. Please note that the Next.js storefront installation time might be long.
5. Once the backend/admin runs, the Next.js storefront will run in the background. You can ensure it's working by opening `localhost:8000` in your browser. The Onboarding flow in the admin should be tailored towards the Next.js starter
6. Terminate the process and check if the Next.js storefront is still running by opening it in your browser again. The storefront should fail to open.

### With Next.js Starter Using Option

1. Run the command (either locally or through the snapshot) and pass it the `--with-nextjs-starter` option.
2. You won't be asked about installing the Next.js starter, but it will be part of the next steps.
3. Follow the same testing steps as the previous section starting from number 3.

### Without Next.js Starter

1. Run the command (either locally or through the snapshot) 
2. When asked about the Next.js starter, answer "N" or just press enter
3. The setup process should continue without installing the Next.js storefront and the onboarding should work without any change.

### CODE REVIEW

File: packages/create-medusa-app/src/utils/clone-repo.ts [hunk 1]
```diff
@@ -18,9 +18,13 @@ export default async function cloneRepo({
   repoUrl,
   abortController,
 }: CloneRepoOptions) {
-  await promiseExec(`git clone ${repoUrl || DEFAULT_REPO} ${directoryName}`, {
-    signal: abortController?.signal,
-  })
+  // TODO remove branch before merging
```
Comment (olivermrbl) [inline packages/create-medusa-app/src/utils/clone-repo.ts]: todo:

File: packages/create-medusa-app/src/utils/nextjs-utils.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import { EOL } from "os"
+
+const NEXTJS_REPO = "https://github.com/medusajs/nextjs-starter-medusa"
+
+export async function askForNextjsStarter(): Promise<boolean> {
+  const { installNextjs } = await inquirer.prompt([
+    {
+      type: "confirm",
+      name: "installNextjs",
+      message: `Would you like to install the Next.js storefront? You can also install it later`,
```
Comment (olivermrbl) [inline packages/create-medusa-app/src/utils/nextjs-utils.ts]: suggestion: Would `create` or `add` make more sense than `install` here? 🤔

**Review by olivermrbl — APPROVED** [review]

File: packages/create-medusa-app/src/utils/nextjs-utils.ts (hunk 2 above)
Comment (shahednasser) [reply packages/create-medusa-app/src/utils/nextjs-utils.ts]: Are we talking about both instances of `install`?

**Pull request merged**